### ISSUE TITLE
Examples

### ISSUE THREAD

**Issue by KottV** [BespokeSynth/BespokeSynth#17]
Hi,
Ryan, could you add some new examples (with scripting too) into data? For example that I saw in your Twitter.
Please)

**Comment by awwbees** [comment]
there are a meager few script examples in https://github.com/awwbees/BespokeSynth/tree/master/Builds/MacOSX/build/Release/data/scripts , which can be loaded via the dropdown on the top of script modules. but yes, overall, example usage is extremely lacking. I'll find some good savestates that don't rely on VSTs to submit to the savestates directory.

**Comment by KottV** [comment]
Yeah, I saw them. Bot when I try to send notes (should I?) in script I get:
```
python execution exception (error_already_set): NameError: name 'on_note__script' is not defined

At:
  <string>(2): <module>
```

**Comment by awwbees** [comment]
ah, first you have to either press the "run" button on the script, or ctrl-R with the cursor in the script window. this executes the script, to define the on_note() function

________________________________

**Comment by KottV** [comment]
I'm feel stupid, no success for me. I press run, script windows blinked green, then I send note from my keys and got that error. For example this simple state:

![image](https://user-images.githubusercontent.com/4686351/83313107-1f0d2b80-a258-11ea-9664-de4d467640e3.png)

**Comment by awwbees** [comment]
ah yes, this is definitely a case where examples would help. the problem in your case is that there is an `on_pulse()` defined, which takes pulse input, but not an `on_note()`, which takes note input. here's a version of your patch that takes pulses in:
<img width="848" alt="Screen Shot 2020-05-29 at 7 33 20 PM" src="https://user-images.githubusercontent.com/839239/83313406-0f53ff80-a1e4-11ea-8acc-4a62e45903ed.png">

and then here's a version that does closer to what you're trying to do, which is manipulate note input:
<img width="1039" alt="Screen Shot 2020-05-29 at 7 36 30 PM" src="https://user-images.githubusercontent.com/839239/83313435-2a267400-a1e4-11ea-99e2-35325025d6a8.png">

in making that note input example, I found that I'm hearing some pops and clicks. mostly I've been using sequencers for note input, which handles the scheduling appropriately, but it seems I've got a bug in handling live input well. it'll probably sound fine with a VST, but the internal synths will all probably give pops and clicks with live input triggered by `on_note()`. I'll take a look at that sometime soon-ish
